apps/astarte_realm_management_api/lib/astarte_realm_management_api/triggers/triggers.ex

```diff
@@ -65,10 +66,18 @@ defmodule Astarte.RealmManagement.API.Triggers do
       simple_triggers_configs =
         Enum.map(tagged_simple_triggers, &SimpleTriggerConfig.from_tagged_simple_trigger/1)
 
+      action_struct =
+        Changeset.cast(
+          %Action{},
+          action_map,
+          Action.__schema__(:fields)
+        )
+        |> Changeset.apply_changes()
```

Comment:
Since we're assuming the `action_map` not to need validation, what about:
```suggestion
    # some lines above
    {:ok, action_map} <- Jason.decode(action, keys: :atoms!) do
      #...
      action_struct = struct(Action, action_map)
```